### PR TITLE
Removed LastKnownLocation for CurrentLocation

### PR DESCRIPTION
**Pull request by CommandrMoose**
This PR removes the LastKnownLocation from the exterior manager and moves it to the Piloting manager under the new name "CurrentLocation".

Made entering a TARDIS update the currentLocation.
Made flight set the current location when the ship lands.
Re-worked @Jeryn99's logic for removing duplicate shells, adding a failsafe so if there's no recorded current value we make that shell the new value.

### CODE REVIEW

File: common/src/main/java/whocraft/tardis_refined/common/tardis/control/flight/ReadoutControl.java
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/control/flight/ReadoutControl.java]: Add Lang translation

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java:188]: Use of specific method ``removeBlock`` may be better user here to be consistent with other instances

**Review by 50ap5ud5 — CHANGES_REQUESTED** [review]
Some minor changes, looking good.

File: common/src/main/java/whocraft/tardis_refined/common/blockentity/shell/ShellBaseBlockEntity.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/blockentity/shell/ShellBaseBlockEntity.java]: getPilotingManager used multiple times, make it a local variable

File: common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java:261]: getCurrentLocation() used multiple times, make it a local variable

File: common/src/main/java/whocraft/tardis_refined/common/tardis/control/flight/RandomControl.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/control/flight/RandomControl.java:29]: Code has existing pilotManager variable yet getter is used

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java]: Code should make pilotingManager a existing variable, null check that and then use it in code

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java]: getPilotingManager used excessively as getter, should be a local variable
getCurrentLocation used excessively as getter, should be a local variable

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java:227]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisInteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisInteriorManager.java]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisInteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisInteriorManager.java]: getExteriorManager should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java:166]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java:126]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java:269]: getPilotingManger should be null checked before accessing it's methods

File: common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java
Comment (Jeryn99) [inline common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java:229]: getPilotingManger should be null checked before accessing it's methods

**Review by Jeryn99 — CHANGES_REQUESTED** [review]
Code from PR and legacy code needs changes before approval

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java:188]: We actually favor setblock over the remove block for most of the codebase.

File: common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/common/capability/TardisLevelOperator.java:261]: It is used once in this instance.

**Comment by CommandrMoose** [comment]
Addressed all the issues mentioned above

**Review by Jeryn99 — APPROVED** [review]

**Review by 50ap5ud5 — APPROVED** [review]

**Pull request merged**
